File: src/main/java/com/jplearning/service/NotificationService.java
package com.jplearning.service;

import com.jplearning.dto.response.NotificationResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface NotificationService {
    /**
     * Create notification for a specific user
     * @param userId ID of the user to notify
     * @param title Notification title
     * @param message Notification message
     * @param type Notification type
     * @param actionUrl Optional action URL
     * @param actionText Optional action text
     */
    void createNotification(Long userId, String title, String message, 
                           com.jplearning.entity.Notification.NotificationType type, 
                           String actionUrl, String actionText);

    /**
     * Create notification for all tutors
     * @param title Notification title
     * @param message Notification message
     * @param type Notification type
     * @param actionUrl Optional action URL
     * @param actionText Optional action text
     */
    void createNotificationForAllTutors(String title, String message, 
                                       com.jplearning.entity.Notification.NotificationType type, 
                                       String actionUrl, String actionText);

    /**
     * Get notifications for a user
     * @param userId ID of the user
     * @param pageable Pagination information
     * @return Page of notification responses
     */
    Page<NotificationResponse> getUserNotifications(Long userId, Pageable pageable);

    /**
     * Mark notification as read
     * @param notificationId ID of the notification
     * @param userId ID of the user
     */
    void markNotificationAsRead(Long notificationId, Long userId);

    /**
     * Mark all notifications as read for a user
     * @param userId ID of the user
     */
    void markAllNotificationsAsRead(Long userId);

    /**
     * Get unread notification count for a user
     * @param userId ID of the user
     * @return Count of unread notifications
     */
    long getUnreadNotificationCount(Long userId);

    /**
     * Clean up old read notifications (scheduled task)
     */
    void cleanupOldNotifications();
} 

File: src/services/notification-service.ts
import api from "@/lib/api";

export interface Notification {
  id: number;
  title: string;
  message: string;
  type: string;
  isRead: boolean;
  actionUrl?: string;
  actionText?: string;
  createdAt: string;
  readAt?: string;
}

export interface NotificationResponse {
  content: Notification[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}

export const NotificationService = {
  // Get user notifications with pagination
  async getMyNotifications(page: number = 0, size: number = 20): Promise<NotificationResponse> {
    const response = await api.get(`/api/notifications/my-notifications?page=${page}&size=${size}`);
    return response.data;
  },

  // Get unread notification count
  async getUnreadCount(): Promise<number> {
    const response = await api.get('/api/notifications/unread-count');
    return response.data;
  },

  // Mark notification as read
  async markAsRead(notificationId: number): Promise<void> {
    await api.put(`/api/notifications/${notificationId}/read`);
  },

  // Mark all notifications as read
  async markAllAsRead(): Promise<void> {
    await api.put('/api/notifications/mark-all-read');
  }
};

export default NotificationService;